src/main.h

```diff
@@ -49,6 +47,28 @@ class CTxUndo;
 struct CNodeStateStats;
 class CTxInUndo;
 
+// Enforce 64-bit architecture requirement
+#if defined(_MSC_VER) || (defined(__INTEL_COMPILER) && defined(_WIN32))
+    #if defined(_M_X64)
+        #define BITNESS_64
+    #else 
+        #define BITNESS_32
+    #endif
+#elif defined(__clang__) || defined(__INTEL_COMPILER) || defined(__GNUC__)
+    #if defined(__x86_64)
+        #define BITNESS_64
+    #else
+        #define BITNESS_32
+    #endif
+#else
+    #error Cannot detect compiler or compiler is not supported
+#endif
+#if defined(BITNESS_32)
+#error "Zend is not supported on 32 bit architecture"
+#endif
+#undef BITNESS_32
+#undef BITNESS_64
+
```

Comment:
Given we know [there are problems on big endian archtectures](https://zcash.readthedocs.io/en/latest/rtd_pages/security_warnings.html#x86-64-linux-only) I believe is worth to add the proper checks here

```suggestion
#if defined(_WIN32)
    // On Windows we assume little-endian
#elif defined(__BYTE_ORDER__)
  #if (__BYTE_ODER__ != __ORDER_LITTLE_ENDIAN__)
    #error "Zend is not supported on big-endian architectures"
  #endif
#else
  #error "Undetectable endianness"
#endif
```

Reply:
Ok, modified as per your suggestion.